internal/controller/controllers/agent_controller.go

```diff
@@ -822,6 +825,41 @@ func (r *AgentReconciler) updateInventory(log logrus.FieldLogger, host *models.H
 	return nil
 }
 
+func updateInventoryLabels(agent *aiv1beta1.Agent) {
+	inventory := agent.Status.Inventory
+	if agent.ObjectMeta.Labels == nil {
+		agent.ObjectMeta.Labels = make(map[string]string)
+	} else {
+		for key := range agent.ObjectMeta.Labels {
+			if strings.HasPrefix(key, FeatureLabelPrefix) {
+				delete(agent.ObjectMeta.Labels, key)
+			}
+		}
+	}
+	if agent.ObjectMeta.Annotations == nil {
+		agent.ObjectMeta.Annotations = make(map[string]string)
+	}
+
+	hasSSD := false
+	for _, d := range inventory.Disks {
+		if d.DriveType == "SSD" {
+			hasSSD = true
+			break
+		}
+	}
+
+	virtEnabled := false
+	if inventory.Cpu.Architecture == common.DefaultCPUArchitecture &&
+		(funk.Contains(inventory.Cpu.Flags, "vmx") || funk.Contains(inventory.Cpu.Flags, "svm")) {
+		virtEnabled = true
+	}
+
+	agent.ObjectMeta.Annotations[FeatureLabelPrefix+"version"] = "0.1"
+	agent.ObjectMeta.Labels[FeatureLabelPrefix+"storage-nonrotationaldisk"] = strconv.FormatBool(hasSSD)
+	agent.ObjectMeta.Labels[FeatureLabelPrefix+"cpu-architecture"] = inventory.Cpu.Architecture
+	agent.ObjectMeta.Labels[FeatureLabelPrefix+"cpu-virtenabled"] = strconv.FormatBool(virtEnabled)
```

Comment:
We should decide what level the labels should be at - we can expose that virt is enabled, or expose the individual CPU flags (this is more in line with NFD and is more flexible but less user-friendly).